tests/5.1/requires/test_target_is_accessible_with_usm.c

```diff
@@ -0,0 +1,50 @@
+//--------------- test_target_is_accessible_with_usm.c---------------------//
+//
+// OpenMP API Version 5.1 Aug 2021
+//
+// This test checks that the omp_target_is_accessible device routine.
+// In this test the output of the target_is_accessible call should return
+// true because the requires unified shared memory is used.
+//-----------------------------------------------------------------------//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 100
+
+#pragma omp requires unified_shared_memory
+
+int check_device(){
+	int errors = 0;
+	int check_test = -57;
+	const int buf_size = sizeof(int) * N;
+	const int dev = omp_get_default_device();
+
+	int *ptr = (int *) malloc(buf_size);
+
+	check_test = omp_target_is_accessible(ptr, buf_size, dev);
+	
+	free(ptr);
+
+	OMPVV_TEST_AND_SET_VERBOSE(errors, check_test == 0);
+
+	OMPVV_ERROR_IF(check_test == -57, "Call to omp_target_is_accessible did not return true or false");
+	OMPVV_ERROR_IF(check_test == 0, "Call to omp_target_is_accessible did not return true when using USM.");
+	OMPVV_INFOMSG_IF((check_test != 0 && check_test != -57), "The host memory is accessible from the default device.");
```

Comment:
maybe switch the OMPVV_INFOMSG_IF to `check_test == 1`, if check_test returns 100 for example, that is still wrong behavior

Reply:
The `==/!= -57` checks looks odd. I think this testcase mangles two things:

* Checking the return value of `omp_target_is_accessible`. This, I think would require something like the following:
```diff
- 	OMPVV_ERROR_IF(check_test == -57, "Call to omp_target_is_accessible did not return true or false");
+ 	OMPVV_ERROR_IF(check_test != 0 && check_test != 1, "Call to omp_target_is_accessible did not return true or false");
```

* Checking whether the memory was really available on the device, I mean something like the following:
```C
for (int i = 0; i < N; i++)
   ptr[i] = -42;
if (check_test)
  {
    !$omp target is_device_ptr(ptr)
       for (int i = 0; i < N; i++)
         ptr[i] = 2*i;
     for (int i = 0; i < N; i++)
       if (ptr[i] /= 2*i)
         ...
  }
OMPVV_INFOMSG_IF(check_test && ptr[1] == 2, "The host memory is accessible from the default device.");
```
or something along this line.

Reply:
@tmh97 @tob2 The first suggestion assumes true == 1, which (maybe is the norm) but not standard . Bool true macro evaluates to 1 and false to 0, but the return value here is an integer.

Even the spec says : "This document uses true as a generic term
for a non-zero integer value and false as a generic term for an integer value of zero."

In essence an implementation could return any non-zero value to indicate true. -57 (a random choice ) is chosen to avoid the more regular values for (int) true.

Your second suggestion is a cross-verification that when the call returns true (as it should for USM) the memory is actually  accessible. IMO it goes slightly beyond the scope of the test, but I am happy to include it.

Reply:
> @tmh97 @tob2 The first suggestion assumes true == 1, which (maybe is the norm) but not standard . Bool true macro evaluates to 1 and false to 0, but the return value here is an integer.

I concur – but have a still a minor remark:

I think `true == 1` matches the C standard (C17 in Sect. 7.18) – but I concur that the `omp_target_is_...` functions do not return `true` but the OpenMP-defined _true_, which can be any nonzero integer. Thus, the suggested `true`/`false` alias `1`/`0` check does not make sense – but likewise, previously used `check_test == -57, "... did not return true or false");` suffers from the same issue.

_Side remark: With Fortran, **omp_target_is_accessible** returns an integer (`integer(c_int)` plus `bind(C)`), hence, it does not return `.true.`/`.false.` – but it neither returns_ true _/_ false _. — That's issue now tracked in OpenMP spec issue 3327._